src/CommandAndStateMessageParser.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef _COMMAND_AND_STATE_MESSAGE_PARSER_HPP_
+#define _COMMAND_AND_STATE_MESSAGE_PARSER_HPP_
+
+#include "base/commands/LinearAngular6DCommand.hpp"
+#include "deep_trekker/DeepTrekkerCommands.hpp"
+#include "deep_trekker/DeepTrekkerStates.hpp"
+#include "power_base/BatteryStatus.hpp"
+#include "stdio.h"
+#include "string.h"
+#include "json/json.h"
+#include <algorithm>
+#include <base/Time.hpp>
+#include <memory>
+
+namespace deep_trekker {
+
+    class CommandAndStateMessageParser {
+    public:
+        CommandAndStateMessageParser();
+
+        std::string parseGetMessage(std::string api_version);
```

Comment:
I'm not asking you to change this ... but for your information

- `parse` is converting a marshalled value (in this case, JSON) into a ready-to-use data structure.
- `marshal` is what converts a data structure into a string representation.